hibernate-core/src/test/java/org/hibernate/orm/test/jpa/callbacks/CallbacksTest.java

```diff
@@ -38,34 +38,40 @@
 		Rythm.class
 })
 public class CallbacksTest {
+
+	@AfterEach
+	public void tearDown(EntityManagerFactoryScope scope) {
+		scope.inTransaction(
+				entityManager -> {
+					entityManager.createQuery( "delete from Cat" ).executeUpdate();
+					entityManager.createQuery( "delete from Television" ).executeUpdate();
+					entityManager.createQuery( "delete from Plant" ).executeUpdate();
+					entityManager.createQuery( "delete from Kitten" ).executeUpdate();
+				}
+		);
+	}
+
 	@Test
 	public void testCallbackMethod(EntityManagerFactoryScope scope) {
-		int id = scope.fromTransaction(
+		scope.inTransaction(
 				entityManager -> {
 					Cat c = new Cat();
 					c.setName( "Kitty" );
 					c.setDateOfBirth( new Date( 90, 11, 15 ) );
 					entityManager.persist( c );
-					return c.getId();
-				}
-		);
-		scope.inTransaction(
-				entityManager -> {
-					Cat _c = entityManager.find( Cat.class, id );
-					assertFalse( _c.getAge() == 0 );
-					_c.setName( "Tomcat" ); //update this entity
-				}
-		);
-		scope.inTransaction(
-				entityManager -> {
-					Cat _c = entityManager.find( Cat.class, id );
-					assertEquals( "Tomcat", _c.getName() );
-				}
-		);
+					entityManager.getTransaction().commit();
 
-		scope.inTransaction(
-				entityManager -> {
-					entityManager.createQuery( "delete from Cat" ).executeUpdate();
+					entityManager.clear();
+
+					entityManager.getTransaction().begin();
+					c = entityManager.find( Cat.class, c.getId() );
+					assertFalse( c.getAge() == 0 );
+					c.setName( "Tomcat" ); //update this entity
+					entityManager.getTransaction().commit();
+					entityManager.clear();
+					entityManager.getTransaction().begin();
+					c = entityManager.find( Cat.class, c.getId() );
+					assertEquals( "Tomcat", c.getName() );
```

Comment:
The point of using separate scope.inTransactions, is precisely to not have to manage the starting, committing, clearing and such manually, which I did only where this is not required for some reason. I don't think it alters the 'spirit' of the test and leaves for much less boilerplate, so why revert back to the 'old' way?

Reply:
because the original test uses the same `Entitymanager` instance for all the test and maybe this is important for the test.

Reply:
and in such a case `scope.inTransaction` avoids writing the code that manages exceptions and rollback the transaction